File: Day-01/python/Amejonah1200/day1.py
def part1(measurements: list[int]) -> int:
    count = 0
    for i in range(len(measurements) - 1):
        if measurements[i] < measurements[i + 1]:
            count += 1
    return count


def part2(measurements: list[int]) -> int:
    return part1([sum(t) for t in zip(measurements, measurements[1:], measurements[2:])])


if __name__ == '__main__':
    with open('../inputs/day1.txt', 'r') as f:
        measurements = [int(line) for line in f.readlines()]
        print("Part1:", part1(measurements))
        print("Part2:", part2(measurements))

File: Day-01/rust/Amejonah1200/day1.rs
pub(crate) fn start() {
  let measurements: Vec<i32> = include_str!("../../inputs/day1.txt").lines().map(|line| line.parse::<i32>().unwrap()).collect();
  println!("Part 1: {}", part1(&measurements));
  println!("Part 2: {}", part2(&measurements));
}

fn part1(measurements: &Vec<i32>) -> usize {
  measurements.windows(2).filter(|pair| pair[0] < pair[1]).count()
}

fn part2(measurements: &Vec<i32>) -> usize {
  part1(&measurements.windows(3).map(|window| window.iter().sum::<i32>()).collect())
}
